src/object_async/hello_async.cpp

```diff
@@ -222,7 +232,7 @@ NAN_METHOD(HelloObjectAsync::helloAsync)
     // the object.
     // - The C++ methods must be static to make them available at startup across
     // the language boundary (JS <-> C++).
-    HelloObjectAsync* h =
+    auto* h =
```

Comment:
@GretaCB - thanks for asking! I'll answer by backing up a bit. A good reason to avoid using `auto` is when using it makes the type hard to quickly understand. Like:

```
auto var = 1.0
auto var2 = var*1;
```

It is not immediately obvious here whether `var2` is a float, double, or integer. This is a case where I don't think `auto` should be used.

But in this case `clang-tidy` automatically fixed the code to use `auto` since it seems like a reasonable use case: it makes the code more concise/shorter but does not obscure the type. Since the type is obvious from the template argument passed to `Unwrap<T>`:

```
auto* h = Nan::ObjectWrap::Unwrap<HelloObjectAsync>(info.Holder());
```

More details at https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html